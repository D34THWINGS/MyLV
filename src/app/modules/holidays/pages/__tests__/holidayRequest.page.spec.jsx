import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { HolidayRequestPage } from '../holidayRequest.page'

jest.unmock('../holidayRequest.page')

describe('HolidayRequestPage', () => {
  let props
  beforeEach(() => {
    props = {
      postHoliday: jest.fn(),
    }
  })

  const getWrapper = () => shallow(<HolidayRequestPage {...props} />)

  it('should render properly', () => {
    // When
    const wrapper = shallow(getWrapper().prop('render')({ children: <div>Foo</div> }))

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
