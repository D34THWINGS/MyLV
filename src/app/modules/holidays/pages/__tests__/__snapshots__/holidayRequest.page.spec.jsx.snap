// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HolidayRequestPage should render properly 1`] = `
<WithStyles(Paper)
  elevation={2}
>
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      component="h2"
      gutterBottom={true}
      variant="headline"
    >
      Nouvelle demande de cong√©
    </WithStyles(Typography)>
    <div>
      Foo
    </div>
  </WithStyles(CardContent)>
  <WithStyles(CardActions)>
    <WithStyles(Button)
      color="primary"
      disabled={true}
      size="small"
      type="submit"
    >
      Envoyer
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</WithStyles(Paper)>
`;
